docs/fluxdoc.md

```diff
@@ -0,0 +1,227 @@
+# fluxdoc formatting
+
+`fluxdoc` uses comments in `.flux` package files in `/stdlib` to generate and
+output standard library documentation in JSON format.
+The generated JSON is used to build the public-facing Flux standard library
+documentation and ensure documentation is up-to-date and featur-complete with
+each new Flux release.
+
+**Note:** Packages are required to pass the `fluxdoc` linter before being merged
+into `master` unless listed as an exception in
+[`/libflux/flux-core/src/bin/fluxdoc.rs`](https://github.com/influxdata/flux/blob/master/libflux/flux-core/src/bin/fluxdoc.rs).
+
+## Syntax and structure
+Each `.flux` package file in `/stdlib` should include comments using the
+following syntax and structure.
+Flux comment lines begin with `//`.
+
+### Use Markdown
+Inline Flux documentation uses Markdown or, to be more specific,
+[GitHub Flavored Markdown (GFM)](https://guides.github.com/features/mastering-markdown/).
```

Comment:
Technically we support CommonMark https://spec.commonmark.org/ with support for Github Tables and tasks list. FWIW we are using https://docs.rs/crate/pulldown-cmark/0.8.0 under the hood.